File: 4/daa/Floyd/p1.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int n=6;
    int** graph=(int **)malloc(7*sizeof(int *));
    int** prev=(int **)malloc(7*sizeof(int *));
    int** next=(int **)malloc(7*sizeof(int *));
    for(int i=0;i<7;i++){
        graph[i]=(int *) malloc(7*sizeof(int));
        prev[i]=(int *) malloc(7*sizeof(int));
        next[i]=(int *) malloc(7*sizeof(int));
    }
    for(int i=0;i<7;i++){
        for(int j=0;j<7;j++){
            if(i==j){
            graph[i][j]=0;
            prev[i][j]=0;
            }
            else{
            graph[i][j]=1000;
            prev[i][j]=1000;}
        }
    }
    graph[4][1]=-4;graph[2][1]=1;graph[3][2]=2;graph[2][4]=2;graph[1][5]=-1;graph[6][2]=5;graph[3][6]=-8;graph[6][3]=10;graph[5][2]=7;graph[4][5]=3;
    prev[4][1]=-4;prev[2][1]=1;prev[3][2]=2;prev[2][4]=2;prev[1][5]=-1;prev[6][2]=5;prev[3][6]=-8;prev[6][3]=10;prev[5][2]=7;prev[4][5]=3;
    //floyd
    for(int k=1;k<=n;k++){
        for(int i=1;i<=n;i++){
            for(int j=1;j<=n;j++){
                next[i][j]=min(prev[i][j],prev[i][k]+prev[k][j]);
            }
            prev=next;
        }
    }
    cout<<"graph is\n";
    for(int i=1;i<7;i++){
        for(int j=1;j<7;j++){
            cout<<"\t"<<graph[i][j];
        }
        cout<<"\n";
    }
    cout<<"min distance\n";
    for(int i=1;i<7;i++){
        for(int j=1;j<7;j++){
            cout<<"\t"<<prev[i][j];
        }
        cout<<"\n";
    }
    return 1;
}

File: 4/daa/Floyd/p1.py
class Graph:
    def __init__(self,n):
        self.n=n
        self.adjacencylist=[[]for i in range(n)]
        self.weight=[[1000 for i in range(n)] for j in range(n)]
    def weight1(self,i,j,w):
        self.weight[i][j]=w
    def floyd(self):
        w=self.weight
        for i in range(n):
            w[i][i]=0
        path=[[[[0]for i in range(n)]for j in range(n)]for k in range(n)]
        for k in range(self.n):
            for i in range(self.n):
                for j in range(self.n):
                    temp=w[i][j]
                    w[i][j]=min(w[i][j],w[i][k]+w[k][j])
                    if(w[i][j]==temp):
                        path[k][i][j]=path[k-1][i][j]
                    else:
                        path[k][i][j]=path[k-1][k-1][j]
                    if(k==self.n-1):
                        print("from",i,"to",j,"dis",w[i][j])
        
def main():
    n=4
    g=Graph(n)
    g.adjacencylist[0].append(1)
    g.adjacencylist[0].append(2)
    g.adjacencylist[1].append(3)
    g.adjacencylist[2].append(1)
    g.adjacencylist[1].append(3)
    g.weight1(0,1,1)
    g.weight1(0,2,3)
    g.weight1(1,2,2)
    g.weight1(3,1,4)
    g.weight1(2,3,1)
    g.floyd()
if __name__ == '__main__':
    main()